src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -961,14 +962,20 @@ private void Advance()
             _ilg.Emit(OpCodes.Br, AdvanceLabel());
         }
 
-        private void CallToLower()
+        private void InitLocalCultureInfo()
         {
             if ((_options & RegexOptions.CultureInvariant) != 0)
                 Call(s_getInvariantCulture);
             else
                 Call(s_getCurrentCulture);
 
-            Call(s_chartolowerM);
+            Stloc(_cultureV);
+        }
+
+        private void CallToLower()
+        {
+             Ldloc(_cultureV);
+             Call(s_chartolowerM);
```

Comment:
@ViktorHofer, am I misremembering, or did you do some ToLower-related optimizations/changes when you were cleaning up the regex code recently?  If yes, should we instead just port the appropriate changes to the compiled version?  If no, is there a similar optimization to be done on the interpreted side of things?

Reply:
Right, but only minor allocation optimizations. I wouldn't do that as part of this PR but revisit the compiled code paths later to make sure that we don't diverge and bring optimizations over.